ds_judgements_public_ui/static/js/src/modules/date_input.js

```diff
@@ -0,0 +1,26 @@
+import $ from "jquery";
+
+function constrainInputToMinMax() {
+    const input = $(this);
+    var lastValidValue = input.value;
+    input.on("input", (event) => {
+        const validity = input[0].validity;
+        if (validity.badInput) {
+            this.value = lastValidValue;
+        } else if (validity.rangeOverflow) {
+            this.value = this.max;
+        } else if (!validity.badInput) {
```

Comment:
Would an `else` be sufficient here since by default `!validity.badInput` is true at this point in the conditional block?

Reply:
aha i think that needs to be "if there are no other errors at all", which isn't ".badInput", it's ... another property who's name escapes me. I'll go back and check if the logic is correct here, and amend if not. If it is correct, will amend as you suggest. Good spot!

Reply:
aha no, you're entirely right. Fixed!